iOS_SDK/OneSignalSDK/Source/OneSignalLocation.m

```diff
@@ -186,15 +187,17 @@ + (void) internalGetLocation:(bool)prompt {
         //Location Always requires: Location Background Mode + NSLocationAlwaysUsageDescription
         NSArray* backgroundModes = [[NSBundle mainBundle] objectForInfoDictionaryKey:@"UIBackgroundModes"];
         NSString* alwaysDescription = [[NSBundle mainBundle] objectForInfoDictionaryKey:@"NSLocationAlwaysUsageDescription"] ?: [[NSBundle mainBundle] objectForInfoDictionaryKey:@"NSLocationAlwaysAndWhenInUseUsageDescription"];
-        if(backgroundModes && [backgroundModes containsObject:@"location"] && alwaysDescription) {
+        // use background location updates if always permission granted or prompt allowed
+        if(backgroundModes && [backgroundModes containsObject:@"location"] && alwaysDescription && (permissionStatus == 3 || prompt)) {
```

Comment:
Just to confirm with the `3` value here you are checking for `kCLAuthorizationStatusAuthorized`?
https://developer.apple.com/documentation/corelocation/clauthorizationstatus?language=objc

I believe we should be checking for `kCLAuthorizationStatusAuthorizedWhenInUse` as well.

Reply:
That is covered by the check if `prompt` is true: `(permissionStatus == 3 || prompt)`.

If the status is `kCLAuthorizationStatusAuthorizedWhenInUse` then background location updates should only be attempted if `prompt`ing is allowed.

Reply:
I see, thanks for confirming.